src/Init/Data/Cast.lean

```diff
@@ -44,25 +44,26 @@ Nat → Nat → ...`. Sometimes we also need to declare the `CoeHTCT`
 instance if we need to shadow another coercion.
 -/
 
-/-- Type class for the canonical homomorphism `Nat → R`. -/
+/--
+The canonical homomorphism `Nat → R`. In most use cases, the target type will have a (semi)ring
+structure, and this homomorphism should be a ring homomorphism.
```

Comment:
```suggestion
structure, and this homomorphism should be a (semi)ring homomorphism.
```

If we're telling people that we expect a semiring, we should ask for a semiring hom :)

Reply:
Thanks!